### PR TITLE
chore: increase type-safety with stricter tsconfig settings

### PR DESCRIPTION
**Pull request by benpsnyder**

### CODE REVIEW

File: libs/ui/toggle/brain/src/lib/brn-toggle.directive.ts [hunk 1]
```diff
@@ -39,15 +38,13 @@ export class BrnToggleDirective {
 	public id = `brn-toggle-${BrnToggleDirective.uniqueId++}`;
 
 	@Input()
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	public value: any;
+	public value: string | number | null = null;
```
Comment (alexciesielski) [inline libs/ui/toggle/brain/src/lib/brn-toggle.directive.ts:41]: Should probably also allow undefined, as Signals default to undefined (as opposed to the async pipe which defaults to null) - there will definitely come a follow up ticket about this.

File: libs/ui/core/src/lib/brain/rx-host-integration.ts [hunk 2]
```diff
@@ -27,6 +27,9 @@ function process<T>(element: HTMLElement, prop: string): (value: T) => void {
 	const [key, unit = ''] = prop.replace('attr.', '').replace('style.', '').replace('class.', '').split('.');
 
 	return (value) => {
+		// Ensure that the key is a valid string
+		if (!key) return;
```
Comment (alexciesielski) [inline libs/ui/core/src/lib/brain/rx-host-integration.ts:31]: nitpick: this is confusing. you're saying to ensure that key is a string, but you're not really checking if it is? is the comment necessary even?

File: libs/ui/toggle/brain/src/lib/brn-toggle.directive.ts (hunk 1 above)
Comment (goetzrobin) [reply libs/ui/toggle/brain/src/lib/brn-toggle.directive.ts:41]: I think this should remain any. It gives developers the most flexibility and we shouldn't dictate what they use as their value. It's also how the Angular CDK works and I wanna keep that

File: apps/app/src/app/pages/(components)/components/(dialog)/dialog-dynamic-component.preview.ts [hunk 3]
```diff
@@ -107,7 +107,7 @@ export class DialogDynamicComponentPreviewComponent {
 	`,
 })
 class SelectUserComponent {
-	@HostBinding('class') private readonly _class: string = 'flex flex-col gap-4';
+	// @HostBinding('class') private readonly _class: string = 'flex flex-col gap-4';
```
Comment (goetzrobin) [inline apps/app/src/app/pages/(components)/components/(dialog)/dialog-dynamic-component.preview.ts:110]: we can move this to the host property

File: libs/cli/src/generators/theme/libs/add-theme-to-application-styles.ts [hunk 4]
```diff
@@ -74,7 +74,7 @@ function findStylesEntryPoint(tree: Tree, project: ProjectConfiguration): string
 	}
 
 	// then check for the specified styles in the build configuration if it exists
-	const styles: Array<string | { input: string; inject: boolean }> = project.targets?.build.options?.styles;
+	const styles: Array<string | { input: string; inject: boolean }> = project.targets?.['build'].options?.styles;
```
Comment (goetzrobin) [inline libs/cli/src/generators/theme/libs/add-theme-to-application-styles.ts:77]:
```suggestion
	const styles: Array<string | { input: string; inject: boolean }> = project.targets?.['build']?.options?.styles;
```
Might not be needed but "seems" logical to keep the chaining all the way through

File: libs/ui/accordion/brain/src/lib/brn-accordion.directive.ts [hunk 5]
```diff
@@ -120,7 +119,7 @@ export class BrnAccordionDirective implements AfterContentInit, OnDestroy {
 	private readonly _el = inject(ElementRef);
 	private _keyManager?: FocusKeyManager<BrnAccordionTriggerDirective>;
 	private _focusMonitor = inject(FocusMonitor);
-	private _ngZone = inject(NgZone);
+	// private _ngZone = inject(NgZone);
```
Comment (goetzrobin) [inline libs/ui/accordion/brain/src/lib/brn-accordion.directive.ts:122]: If it's not needed should we delete this?

File: libs/ui/avatar/brain/src/lib/image/brn-avatar-image.directive.ts [hunk 6]
```diff
@@ -8,15 +8,15 @@ import { Directive, HostListener, computed, signal } from '@angular/core';
 export class BrnAvatarImageDirective {
 	private readonly loaded = signal(false);
 
-	@HostListener('error')
```
Comment (goetzrobin) [inline libs/ui/avatar/brain/src/lib/image/brn-avatar-image.directive.ts:11]: Feels like this needs to be moved to host in the @Directive also?

File: libs/ui/checkbox/helm/src/lib/hlm-checkbox.component.ts [hunk 7]
```diff
@@ -83,12 +82,12 @@ export class HlmCheckboxComponent {
 	protected readonly _disabled = signal(false);
 	public readonly disabled = input(false, { transform: booleanAttribute });
 
-	private disableInput = effect(
```
Comment (goetzrobin) [inline libs/ui/checkbox/helm/src/lib/hlm-checkbox.component.ts:86]: we can move this to the constructor to avoid the unused variable

**Comment by goetzrobin** [comment]
Closing this for now as we go with eslint/prettier until all features are supported

**Pull request closed without merging**